Solutions/Marain.TenantManagement.Abstractions/Marain/TenantManagement/Configuration/ConfigurationItemExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// <copyright file="ConfigurationItemExtensions.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.TenantManagement.Configuration
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Marain.TenantManagement.Exceptions;
+    using Marain.TenantManagement.ServiceManifests;
+
+    /// <summary>
+    /// Extension methods for the <see cref="ConfigurationItem"/> class.
+    /// </summary>
+    public static class ConfigurationItemExtensions
+    {
+        /// <summary>
+        /// Validates the list of <see cref="ConfigurationItem"/> and throws an
+        /// <see cref="InvalidConfigurationException"/> if any errors are encountered.
+        /// </summary>
+        /// <param name="configurationItems">The configuration items to validate.</param>
+        /// <exception cref="InvalidConfigurationException">The configuration items contain one or more errors.</exception>
+        public static void ValidateAndThrow(
+            this ConfigurationItem[] configurationItems)
```

Comment:
Is there a reason for this to be an array, and not `IEnumerable<ConfigurationItem>`?

Reply:
Updated